pytest_tests/testsuites/object/test_object_lock.py

```diff
@@ -663,3 +663,29 @@ def test_link_object_of_complex_object_should_also_be_protected_from_deletion(
                     self.shell,
                     self.cluster.default_rpc_endpoint,
                 )
+
+    @allure.title("Locked object should be protected from deletion after the storage nodes are restarted")
+    @pytest.mark.parametrize(
+        # Only complex objects are required for this test
+        "locked_storage_object",
+        [pytest.lazy_fixture("complex_object_size")],
+        indirect=True,
+    )
+    def test_the_object_lock_should_be_kept_after_restarting_the_nodes(
+            self,
+            locked_storage_object: StorageObjectInfo,
+    ):
+        """
+        Lock objects should fill metabase on resync_metabase
+        """
+        restart_storage_nodes(self.cluster.storage_nodes)
```

Comment:
This is not enough for the nspcc-dev/neofs-node#1502 case, metabase must be resynchronized.

Reply:
> This is not enough for the [nspcc-dev/neofs-node#1502](https://github.com/nspcc-dev/neofs-node/issues/1502) case, metabase must be resynchronized.

@roman-khimov @carpawell  can you please recommend what is the way to make the metabase resynchronized?

Reply:
I strongly recommend dropping metabase too.

Reply:
And there is a `delete_storage_node_data()` for it already in the testlib.

Reply:
@roman-khimov @carpawell If I physically delete the metabase files, should I set resync_metabase to true?
Or should I change the configuration file in the dev-env https://github.com/nspcc-dev/neofs-dev-env/blob/master/services/storage/cfg/config.yml#L39 ?
Why is resync_metabase false?

Reply:
> If I physically delete the metabase files, should I set resync_metabase to true?

Yes.

> Why is resync_metabase false?

It's never `true` in production, this doesn't make much sense, it's only needed when there are some problems with it (storage failure or meta-incompatible node upgrade). So dev-env should have it set to `false` as usual, it's only some specific test that wants to do something. BTW, maybe it'd be easier for you to use environment variable to override this setting.

Reply:
After deleting the metabase files from all node object deletion fails with an error:
`gRPC dial: context deadline exceeded`

https://http.t5.fs.neo.org/86C4P6uJC7gb5n3KkwEGpXRfdczubXyRNW5N9KeJRW73/334-1694095194/index.html#suites/b34feeefa7bd0ee424f598738938f10a/afc0e4a3ea957e70/

@roman-khimov @carpawell please tell me, is this expected behavior?

Reply:
this is a too wide error, usually, it means that a node is not ready to handle connections (not started successfully?)

Reply:
Fixed